### PR TITLE
intro: add heterogeneous tree class

### PR DESCRIPTION
**Pull request by davidkostov**
Adding the HeterogeneousTree class and testing it in the demo class

### CODE REVIEW

File: src/com/clouway/into/HeterogeneousTree.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    }
+
+    /**
+     * This method is used for printing the elements of the tree(pre-order)
+     *
+     * @param focalElement is the current element we are on in the iteration
+     * @param <T> is the type of the element
+     */
+    public <T>void printElements(Node<T> focalElement){
```
Comment (mgenov) [inline src/com/clouway/into/HeterogeneousTree.java]: Node is defined as private class but printElements is a public member which accepts private class. This smells. 

tree.printElements() looks better as the client code will not pass Node.

File: src/com/clouway/into/HeterogeneousTree.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    }
+
+    /**
+     * A method used to get the root of the tree object
+     *
+     * @return the root of the tree
+     */
+
+    public Node getRoot(){
```
Comment (mgenov) [inline src/com/clouway/into/HeterogeneousTree.java]: Root node should not be visible for the calls as it's an internal thing.

File: src/com/clouway/into/HeterogeneousTree.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
+package com.clouway.into;
+
+/**
+ * @author David Kostov (david.kostov.cw@gmail.com)
+ */
+public class HeterogeneousTree{
+
+    //The root variable will contain the root of the tree
+    private Node root;
```
Comment (mgenov) [inline src/com/clouway/into/HeterogeneousTree.java]: Node is defined as template class and it should be used like that. 

```java
Node<Object> 
```

The usage should lead you to something else.

Comment (davidkostov) [inline src/com/clouway/into/HeterogeneousTree.java]: Fixed.

File: src/com/clouway/into/HeterogeneousTree.java (hunk 1 above)
Comment (davidkostov) [inline src/com/clouway/into/HeterogeneousTree.java]: Fixed.

File: src/com/clouway/into/HeterogeneousTree.java (hunk 2 above)
Comment (davidkostov) [inline src/com/clouway/into/HeterogeneousTree.java]: Fixed.

**Comment by davidkostov** [comment]
Fixed.

**Pull request merged**